percona-xtradb-cluster/5.6/source/release-notes/Percona-XtraDB-Cluster-5.6.22-25.8.rst

```diff
@@ -0,0 +1,67 @@
+
+ :variable:`gcache.mem_size` has been deprecated. A warning will now be generated if the variable has value different than ``0``. Bug fixed :bug:`1392408`.
+
+ ``stderr`` of SST/Innobackupex is logged to syslog with appropriate tags if ``sst-syslog`` is in ``[sst]`` or ``[mysqld_safe]`` has syslog in :file:`my.cnf`. This can be overridden by setting the :variable:`sst-syslog` to ``-1`` in ``[sst]``. Bug fixed :bug:`1399134`.
+
+ ``clustercheck`` can now check if the node is ``PRIMARY`` or not, to allow for synced nodes which go out of ``PRIMARY`` not to take any writes/reads. Bug fixed :bug:`1403566`.
+
+ |SST| will now fail early if the :file:`xtrabackup_checkpoints` is missing on the joiner side. Bug fixed :bug:`1405985`.
+
+ ``socat`` utility was not properly terminated after a timeout. Bug fixed :bug:`1409710`.
+
+ When started (without bootstrap), the node would hang if it couldn't find a primary node. Bug fixed :bug:`1413258`.
+
+ 10 seconds timeout in :ref:`xtrabackup_sst` script was not enough for the joiner to delete existing files before it started the socat receiver on systems with big ``datadir``. Bug fixed :bug:`1413879`.
+
+ Non booststrap node could crash while attempting to perform ``table%cache`` operations with the ``BF applier failed to open_and_lock_tables`` warning. Bug fixed :bug:`1414635`.
+
+ |Percona XtraDB Cluster| 5.6 would crash on ``ALTER TABLE`` / ``CREATE INDEX`` with ``Failing assertion: table->n_rec_locks == 0``. Bug fixed :bug:`1282707`.
+
+ Variable length arrays in WSREP code were causing debug builds to fail. Bug fixed :bug:`1409042`.
+
+ Race condition between donor and joiner in :ref:`xtrabackup_sst` has been fixed. This caused :ref:`XtraBackup SST <xtrabackup_sst>` to fail when joiner took longer to spawn the second listener for SST. Bug fixed :bug:`1405668`.
+
+ Signal handling in ``mysqld`` has been fixed for SST processes. Bug fixed :bug:`1399175`.
+
+ SST processes are now spawned with ``fork/exec`` instead of ``posix_spawn`` to allow for better cleanup of child processes in event of non-graceful termination (``SIGKILL`` or a crash etc.). Bug fixed :bug:`1382797`.
+
+ :variable:`wsrep_local_cached_downto` would underflow when the node on which it is queried had no writesets in gcache. Bug fixed :bug:`1262179`.
+ 
+ A typo in :variable:`wsrep_provider_options` could cause an unhandled exception. Bug fixed :githubbug:`215`.
+
+ Interrupted IST would result in ``HA_ERR_KEY_NOT_FOUND`` error in subsequent IST. Bug fixed :githubbug:`210`.
+
+ ``garb`` logger was showing incorrect error message. Bug fixed :githubbug:`168`.
+
+Other bugs fixed: :bug:`1275814`. 
+
+Known Issues
+============
+
+For those affected by crashes on donor during SST due to backup locks (:bug:`1401133`), please add the following to your :file:`my.cnf` configuration file: ::
+
+  [sst]
+  inno-backup-opts='--no-backup-locks'
```

Comment:
Quick addition here, it is also possible to force FTWRL in place of backup locks through environment variable FORCE_FTWRL=1. 

This was added for testing purposes but I suppose others also  can use it. 

For passing environment variables, users can use (quoted from previous release notes):
```
Percona XtraDB Cluster now reads environment variables for mysqld from following files (if present): /etc/default/mysql in Debian/Ubuntu; /etc/sysconfig/mysql in CentOS 6 or lower; /etc/sysconfig/mysql in CentOS 7 with mysql.service; /etc/sysconfig/XYZ in CentOS 7 with mysql@XYZ.service (/etc/sysconfig/bootstrap is supplied by default). Bug fixed #1381492.
```